src/mlpack/methods/pca/decomposition_policies/quic_svd_method.hpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file quic_svd_method.hpp
+ * @author Marcus Edel
+ *
+ * Implementation of the QUIC-SVD policy for use in the Principal Components
+ * Analysis method.
+ */
+
+#ifndef MLPACK_METHODS_PCA_DECOMPOSITION_POLICIES_QUIC_SVD_METHOD_HPP
+#define MLPACK_METHODS_PCA_DECOMPOSITION_POLICIES_QUIC_SVD_METHOD_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/quic_svd/quic_svd.hpp>
+
+namespace mlpack {
+namespace pca {
+
+/**
+ * Implementation of the QUIC-SVD policy.
+ */
+class QUICSVDPolicy
+{
+  public:
+
+  /**
+   * Use QUIC-SVD method to perform the principal components analysis (PCA).
+   *
+   * @param epsilon Error tolerance fraction for calculated subspace.
+   * @param delta Cumulative probability for Monte Carlo error lower bound.
+   */
+  QUICSVDPolicy(const double epsilon = 0.03, const double delta = 0.1) :
+       epsilon(epsilon),
+       delta(delta)
+  {
+    /* Nothing to do here */
+  }
+
+  /**
+   * Apply Principal Component Analysis to the provided data set using the
+   * QUIC-SVD method.
+   *
+   * @param data Data matrix.
+   * @param data Centered data matrix.
+   * @param transformedData Matrix to put results of PCA into.
+   * @param eigVal Vector to put eigenvalues into.
+   * @param eigvec Matrix to put eigenvectors (loadings) into.
```

Comment:
The parameter name are different, one called eigvec, and another called coeff.
Besides, The Apply function of PCA, the comments are written for the same functions and different with each other.

pca.hpp

```
 /**
   * Apply Principal Component Analysis to the provided data set.  It is safe to
   * pass the same matrix reference for both data and transformedData.
   *
   * @param data Data matrix.
   * @param transformedData Matrix to put results of PCA into.
   * @param eigval Vector to put eigenvalues into.
   * @param eigvec Matrix to put eigenvectors (loadings) into.
   */
  void Apply(const arma::mat& data,
             arma::mat& transformedData,
             arma::vec& eigval,
arma::mat& eigvec);

```

pca_impl.hpp

```
/**
 * Apply Principal Component Analysis to the provided data set.
 *
 * @param data - Data matrix
 * @param transformedData - Data with PCA applied
 * @param eigVal - contains eigen values in a column vector
 * @param coeff - PCA Loadings/Coeffs/EigenVectors
 */
template<typename DecompositionPolicy>
void PCA<DecompositionPolicy>::Apply(const arma::mat& data,
                arma::mat& transformedData,
                arma::vec& eigVal,
arma::mat& coeff)
```